### DISCUSSION TITLE
Working with geometry on the frontend

### DISCUSSION THREAD

**Discussion by mato5** [iTwin/itwinjs-core#1199] [Q&A] [answered]
Hey everyone,

I'm working with a road model and I'm trying to calculate certain geometry properties on the front-end. It involves for example calculating distances between linear elements or finding points lying on these elements. I'm asking for help how to tackle this issue. As far as I know, the best way to tackle this would be to represent these linear elements as Curve/Path objects and then use their methods to calculate everything. However, with the model I'm provided, I've been only able to query for their respective geometry streams in binary format which I'm afraid is not the right approach to obtain a Path object.

This is a piece of code I'm using to query those geometryStreams:
`const geometryRows = iModelConnection.query("SELECT * FROM bis.geometricelement3d WHERE userLabel LIKE '%3d linear%'");`

I thank you in advance for any help you can provide!

**Comment by bbastings** [comment]
The right approach is to use GeometryStreamIterator in backend code to get the geometric primitives.

**Reply by mato5** [reply]
Then am I correct to assume there is no way to do this on the frontend? Perhaps a way to convert a binary GeometryStream to a JSON one?

**Reply by pmconne** [reply]
You are correct - there is currently no way to obtain a usable geometry stream on the frontend.
Do you have your own backend? If not, what backend are you using?

**Reply by mato5** [reply]
I am using  **@bentley/itwin-viewer-react** npm package with general purpose backend (default one).

**Reply by pmconne** [reply]
We'll add a way to obtain the geometry from the frontend.

**Comment by pmconne** [comment, marked as answer]
@mato5 You can now query geometry from the frontend using [IModelConnection.Elements.loadProps](https://github.com/imodeljs/imodeljs/blob/b6771dcdcfbcb2bf7bc902812f7302379b4d929e/core/frontend/src/IModelConnection.ts#L823).
Details in [changelog](https://github.com/imodeljs/imodeljs/blob/master/docs/changehistory/NextVersion.md#obtaining-element-geometry-on-the-frontend).
These changes will be available in the 2.16.0-dev.10 packages.
